test/TensorFlowRuntime/codable.swift

```diff
@@ -0,0 +1,54 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+// REQUIRES: swift_test_mode_optimize
+// REQUIRES: objc_interop
+
+import TensorFlow
+import Foundation
+import TensorFlowUnittest
+import StdlibUnittest
+
+var CodableTests = TestSuite("TensorFlowCodable")
+
+private func _testRoundTrip<T>(
+  of value: T, expectedJSON json: Data? = nil
+) where T : Codable, T : Equatable {
+  var payload: Data! = nil
+  do {
+    let encoder = JSONEncoder()
+    payload = try encoder.encode(value)
+  } catch {
+    expectUnreachable("Failed to encode \(T.self) to JSON: \(error)")
+  }
+
+  if let expectedJSON = json {
+    expectEqual(expectedJSON, payload, "Produced JSON not identical to expected JSON.")
+  }
+
+  do {
+    let decoder = JSONDecoder()
+    let decoded = try decoder.decode(T.self, from: payload)
+    expectEqual(decoded, value, "\(T.self) did not round-trip to an equal value.")
+  } catch {
+    expectUnreachable("Failed to decode \(T.self) from JSON: \(error)")
+  }
+}
+
+// NOTE: This test currently crashes with:
+// Compiler bug -- Tensor op builtin __tfop_Shape,$in,T,out_type cannot be lowered to LLVM IR
```

Comment:
Try making tensor implementations `@inline(__always)`.

Reply:
That didn't fix the crash, unfortunately.
```swift
// codable.swift
import TensorFlow
import Foundation

func testTensor() throws {
  let tensor = Tensor<Int32>(shape: [2, 3], scalars: Array(1...6))
  let encoder = JSONEncoder()
  let data = try encoder.encode(tensor)
}
try testTensor()
```
```console
$ swift -Xllvm -tf-dump-intermediates -O codable.swift
```

Reply:
It is the case that partitioning won't work on throwing functions yet?